ansible_collections/arista/avd/plugins/plugin_utils/eos_designs_shared_utils/wan.py

```diff
@@ -41,6 +41,18 @@ def wan_role(self: SharedUtils) -> str | None:
             raise AristaAvdError("'wan_role' client requires 'evpn_role' client.")
         return wan_role
 
+    @cached_property
```

Comment:
so not related to your PR but line 31 above:
```
if self.underlay_router is False or self.wan_mode is None:
```

`self.wan_mode` has a default value and will never be None

could you take advantage of this PR to replace it with

```
if self.underlay_router is False:
```

Reply:
done